modules/juce_audio_plugin_client/VST/juce_VST_Wrapper.cpp

```diff
@@ -836,19 +836,26 @@ class JuceVSTWrapper  : public AudioProcessorListener,
 
     void parameterGestureChanged (int, bool) override {}
 
-    void audioProcessorChanged (AudioProcessor*) override
+    void audioProcessorChanged (AudioProcessor*, int flags) override
     {
-        vstEffect.initialDelay = processor->getLatencySamples();
+        lastChangeFlags = flags;
+        if (flags & AudioProcessorListener::Flags::latencyUpdate)
+            vstEffect.initialDelay = processor->getLatencySamples();
         triggerAsyncUpdate();
     }
 
     void handleAsyncUpdate() override
     {
         if (hostCallback != nullptr)
         {
-            hostCallback (&vstEffect, Vst2::audioMasterUpdateDisplay, 0, 0, nullptr, 0);
-            hostCallback (&vstEffect, Vst2::audioMasterIOChanged,     0, 0, nullptr, 0);
+            // VST2 doesn't support updating parameters so parametersUpdate flag will be ignored.
+
+            if (lastChangeFlags & AudioProcessorListener::Flags::programUpdate)
+                hostCallback (&vstEffect, Vst2::audioMasterUpdateDisplay, 0, 0, nullptr, 0);
+            if (lastChangeFlags & AudioProcessorListener::Flags::latencyUpdate)
+                hostCallback (&vstEffect, Vst2::audioMasterIOChanged, 0, 0, nullptr, 0);
         }
+        lastChangeFlags = 0;

```

Comment:
Hmm this is tricky. 
Is it possible for `audioProcessorChanged` to be called a few times with different flags, -before- the handling of the first one (or just one handle for all of them, didn't dig into how it works)? If so, then the current implementation might "miss" flags which are present in the earlier calls, but not in the last one.

Reply:
Theoretically it is possible to 'bombard' this, getting the `lastChangeFlags` missing callbacks.
But:

I didn't see a case where you'd call this repeatedly that much and didn't want to add a lock because I didn't saw such a call and this might be called from audio thread.

Reply:
I think this is solveable without a lock. You just have to atomically add each new change, and atomically clear each change when you apply it.

Reply:
So are you suggesting atomic queue?
Or simply compare exchange before/after?

Reply:
Simpler, just use `|=` and `&=`  https://en.cppreference.com/w/cpp/atomic/atomic/operator_arith2

Reply:
Ahh! so simply since those are flags you're suggesting worse case just coalesce them. right?

Reply:
Yes exactly.